tests/unit/test_docs.py

```diff
@@ -0,0 +1,87 @@
+"""Test for the sphinx documentation."""
+import subprocess
+import glob
+import os
+import pytest
+
+UNDOCUMENTED_FILES = ["__init__", "constants", "lib_mapper", "protocol_mapper", "variables"]
+
+SPHINX_DIRECTORIES = [
+    {
+        "source_dir": "docs/source/",
+        "build_dir": "docs/build/",
+    }
+]
+
+MODULE_FOLDERS = [{"doc_module_folder": "../../docs/source/netutils/", "netutils_module_folder": "../../netutils/"}]
+
+
+@pytest.fixture
+def load_readme_into_lines():
+    """Loads README.md file into a list of lines."""
+    with open("README.md", "r") as file:
+        return file.readlines()
+
+
+@pytest.mark.parametrize("data", SPHINX_DIRECTORIES)
+def test_sphinx_build(data):
+    sphinx_dummy_build = subprocess.run(  # pylint: disable=W1510
+        ["sphinx-build", "-b", "dummy", "-W", data["source_dir"], data["build_dir"]], stdout=subprocess.PIPE
+    )
+    assert sphinx_dummy_build.returncode == 0
+
+
+@pytest.mark.parametrize("data", MODULE_FOLDERS)
+def test_folders_present_for_module(data):
+    netutils_modules = [file[15:-3] for file in glob.glob(data["netutils_module_folder"] + "*.py")]
+    netutils_modules = list(set(netutils_modules).difference(set(UNDOCUMENTED_FILES)))
+    doc_module_folders = [folder[27:-1] for folder in glob.glob(data["doc_module_folder"] + "*/")]
+
+    for module in netutils_modules:
+        assert module in doc_module_folders
+
+
+@pytest.mark.parametrize("data", MODULE_FOLDERS)
+def test_folders_contain_index(data):
+    doc_module_folders = glob.glob(data["doc_module_folder"])
+
+    for folder in doc_module_folders:
+        assert "index.rst" in os.listdir(folder)
+
+
+def test_overview_section_start_end_lines(load_readme_into_lines):  # pylint: disable=redefined-outer-name
```

Comment:
Can we just create a data structure, and a single test. Something like 

```
[
  {
    "overview": {
      "start-line": 10,
      "start-val": "# netutils\n",
      "end-line": 32,
      "end-value": "* VLANs - Provide the ability to convert configuration into lists or lists into configuration.\n"
    }
  },
  {
    "attribution": {
      "start-line": "NN",
      "start-val": "TEXT",
      "end-line": "NN",
      "end-value": "TEXT"
    }
  }
]
```

Even better would be if you can dynamically find the line numbers, that are shown here, as an example:
```
   :start-line: 131   :end-line: 178
```

Reply:
Updated. I created an issue for parsing the `start` and `end` line so I can get that handled down the road.